sw/device/lib/base/bitfield.h

```diff
@@ -13,58 +14,101 @@ extern "C" {
 #endif  // __cplusplus
 
 /**
- * Masked field offset for 32-bit bitfields, with optional value.
+ * A field of a 32-bit bitfield.
  *
- * `index` represents a shift in bits starting from the 0 bit of a given 32-bit
- * bitfield. It is used to zero the memory inside this bitfield by applying an
- * inverted `mask` at the `index` offset. `value` is then capped by applying
- * `mask` and is shifted to the `index` bits of the bitfield.
+ * The following field definition: `{ .mask = 0xb11, .index = 12 }`
  *
- * Example:
- * ```
- * 32-bit bitfield = 0b11111111'11111111'11111111'11111111
- * set_field32( mask = 0b11111111, index = 16, value = 0b01111110 (126) )
- * result 32-bit bitfield = 0b11111111'01111110'11111111'11111111
- * ```
+ * Denotes the X-marked bits in the following 32-bit bitfield:
+ *
+ *     field:  0b--------'--------'--XX----'--------
+ *     index:   31                                 0
+ *
+ * Restrictions: The index plus the width of the mask must not be greater than
+ * 31.
  */
 typedef struct bitfield_field32 {
-  uint32_t mask;  /**< The field mask. */
-  uint32_t index; /**< The field position within a bitfield. */
-  uint32_t
-      value; /**< The field value to be written, if using a `set` function. */
+  /** The field mask. Usually all ones. */
+  uint32_t mask;
+  /** The field position in the bitfield, counting from the zero-bit. */
+  uint32_t index;
```

Comment:
Not sure it was worth the work to split out the `value` field to then fix up all of the DIFs that use it. But I guess you have already converted the most of the DIFs.

Reply:
Agreed.